build/Dockerfile

```diff
@@ -1,45 +1,262 @@
+	&& yum install -y nginx-${NGINX_VERSION}
+
+
+######### Base image for Openshift with NGINX Plus #########
+FROM openshift-base AS openshift-plus
+
+ENV NGINX_PLUS_VERSION 22-1.el8.ngx
+ARG IC_VERSION
+
+# Install NGINX Plus
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	&& yum install -y wget ca-certificates \
+	&& wget https://nginx.org/keys/nginx_signing.key \
+	&& \
+	NGINX_GPGKEY=nginx_signing.key; \
+	found=''; \
+	for server in \
+		ha.pool.sks-keyservers.net \
+		hkp://p80.pool.sks-keyservers.net:80 \
+		pgp.mit.edu \
+	; do \
+		echo "Fetching GPG key $NGINX_GPGKEY from $server"; \
+		rpm --define="%_hkp_keyserver $server" --import $NGINX_GPGKEY && found=yes && break; \
+	done; \
+	test -z "$found" && echo >&2 "error: failed to fetch GPG key $NGINX_GPGKEY" && exit 1; \
+	echo "[nginx-plus]" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "name=nginx-plus repo" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "baseurl=https://plus-pkgs.nginx.com/centos/8/\$basearch/" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientcert=/etc/ssl/nginx/nginx-repo.crt" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "sslclientkey=/etc/ssl/nginx/nginx-repo.key" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "gpgcheck=1" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& echo "enabled=1" >> /etc/yum.repos.d/nginx-plus-8.repo \
+	&& yum install -y nginx-plus-${NGINX_PLUS_VERSION}
+
+
+######### Create common files, permissions and setcap #########
+FROM $BUILD_OS as files
+
+ARG PLUS
 
 RUN mkdir -p /var/lib/nginx \
 	&& mkdir -p /etc/nginx/secrets \
 	&& mkdir -p /etc/nginx/stream-conf.d \
-	&& apt-get update \
-	&& apt-get install -y libcap2-bin \
 	&& setcap 'cap_net_bind_service=+ep' /usr/sbin/nginx \
 	&& setcap 'cap_net_bind_service=+ep' /usr/sbin/nginx-debug \
 	&& chown -R nginx:0 /etc/nginx \
 	&& chown -R nginx:0 /var/cache/nginx \
 	&& chown -R nginx:0 /var/lib/nginx \
-	&& apt-get remove --purge -y libcap2-bin \
-	&& rm /etc/nginx/conf.d/* \
-	&& rm -rf /var/lib/apt/lists/*
+	&& rm /etc/nginx/conf.d/*
 
-COPY internal/configs/version1/nginx.ingress.tmpl \
-	internal/configs/version1/nginx.tmpl \
-	internal/configs/version2/nginx.virtualserver.tmpl \
-	internal/configs/version2/nginx.transportserver.tmpl /
+COPY internal/configs/version1/nginx$PLUS.ingress.tmpl \
+	internal/configs/version1/nginx$PLUS.tmpl \
+	internal/configs/version2/nginx$PLUS.virtualserver.tmpl \
+	internal/configs/version2/nginx$PLUS.transportserver.tmpl /
 
 # Uncomment the line below if you would like to add the default.pem to the image
 # and use it as a certificate and key for the default server
 # ADD default.pem /etc/nginx/secrets/default
 
-USER nginx
+EXPOSE 80 443
 
 ENTRYPOINT ["/nginx-ingress"]
 
 
-FROM base AS local
-COPY nginx-ingress /
+######### Cleanup for Debian #########
+FROM files as cleanup-debian
```

Comment:
are separate clean up stages necessary?
because we're doing clean up in a  separate layer, cleaning up won't make much difference -- because the files are already stored in the previous layer and thus removing them will not affect the size of the image